### PR TITLE
Added headlines endpoint in News module

### PR DESCRIPTION
**Pull request by shenoac**
Added headlines endpoint.
Added validation for country, category, and limit parameters in headlines endpoint.
Transformed API response to include id, title, description, url, source, and publishedAt.
Updated routes to reference the new schema and ensured proper request validation.
Tested successfully with curl to confirm expected functionality and response format. 
Passing all workflows on forked repo.
Adding as draft until workflows can be run on this repo.

### CODE REVIEW

File: src/modules/News/controller.ts [hunk 1]
```diff
@@ -13,4 +13,41 @@ const getLatestNews = async (req: Request, res: Response) => {
...
+        article: {
+          title: unknown;
+          description: unknown;
+          url: unknown;
+          source: { name: unknown };
+          publishedAt: unknown;
+        },
+        index: number,
+      ) => ({
+        id: `news${index + 1}`,
```
Comment (ahzia) [inline src/modules/News/controller.ts:36]: Suggestion: double check if NEWS API is returning an ID, if yes, return that.

**Review by ahzia — COMMENTED** [review]
Good job @shenoac, take a look at my inline comments.

File: src/modules/News/controller.ts (hunk 1 above)
Comment (shenoac) [reply src/modules/News/controller.ts:36]: Checked wikipedia there is a standard for 2 character country codes

**Review by ahzia — APPROVED** [review]
Good job @shenoac 
I am merging your code.

**Pull request merged**